sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Storage/BlobsCheckpointStore.cs

```diff
@@ -151,16 +191,26 @@ public override async Task<IEnumerable<PartitionOwnership>> ClaimOwnershipAsync(
                     {
                         blobRequestConditions.IfMatch = new ETag(ownership.ETag);
 
-                        try
+                        Func<CancellationToken, Task<Response<BlobInfo>>> overwriteBlobAsync = async uploadToken =>
                         {
-                            infoResponse = await blobClient.SetMetadataAsync(metadata, blobRequestConditions).ConfigureAwait(false);
-                        }
-                        catch (RequestFailedException ex) when (ex.ErrorCode == BlobErrorCode.ContainerNotFound || ex.ErrorCode == BlobErrorCode.BlobNotFound)
+                            try
+                            {
+                                return await blobClient.SetMetadataAsync(metadata, blobRequestConditions, uploadToken);
+                            }
+                            catch (RequestFailedException ex) when (ex.ErrorCode == BlobErrorCode.BlobNotFound)
+                            {
+                                // No ownership was found, which means the ETag should have been set to null in order to
```

Comment:
These are super helpful for those of us not intimately familiar with the algorithm being applied.   Thanks!